community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/cache/NumberArrayFactory.java

```diff
@@ -276,76 +276,19 @@ private Throwable error( long length, int itemSize, Throwable error )
      * some can live on heap and some off heap.
      */
-    NumberArrayFactory CHUNKED_FIXED_SIZE = new Adapter()
-    {
-        private final NumberArrayFactory delegate = new Auto( OFF_HEAP, HEAP );
-
-        @Override
-        public LongArray newLongArray( long length, long defaultValue, long base )
-        {
-            // Here we want to have the property of a dynamic array which makes some parts of the array
-            // live on heap, some off. At the same time we want a fixed size array. Therefore first create
-            // the array as a dynamic array and make it grow to the requested length.
-            LongArray array = newDynamicLongArray( fractionOf( length ), defaultValue );
-            array.at( length - 1 );
-            return array;
-        }
-
-        @Override
-        public IntArray newIntArray( long length, int defaultValue, long base )
-        {
-            // Here we want to have the property of a dynamic array which makes some parts of the array
-            // live on heap, some off. At the same time we want a fixed size array. Therefore first create
-            // the array as a dynamic array and make it grow to the requested length.
-            IntArray array = newDynamicIntArray( fractionOf( length ), defaultValue );
-            array.at( length - 1 );
-            return array;
-        }
-
-        @Override
-        public ByteArray newByteArray( long length, byte[] defaultValue, long base )
-        {
-            // Here we want to have the property of a dynamic array which makes some parts of the array
-            // live on heap, some off. At the same time we want a fixed size array. Therefore first create
-            // the array as a dynamic array and make it grow to the requested length.
-            ByteArray array = newDynamicByteArray( fractionOf( length ), defaultValue );
-            array.at( length - 1 );
-            return array;
-        }
-
-        private long fractionOf( long length )
-        {
-            int maxArraySize = Integer.MAX_VALUE - Short.MAX_VALUE;
-            return min( length / 10, maxArraySize );
-        }
-
-        @Override
-        public IntArray newDynamicIntArray( long chunkSize, int defaultValue )
-        {
-            return new DynamicIntArray( delegate, chunkSize, defaultValue );
-        }
-
-        @Override
-        public LongArray newDynamicLongArray( long chunkSize, long defaultValue )
-        {
-            return new DynamicLongArray( delegate, chunkSize, defaultValue );
-        }
-
-        @Override
-        public ByteArray newDynamicByteArray( long chunkSize, byte[] defaultValue )
-        {
-            return new DynamicByteArray( delegate, chunkSize, defaultValue );
-        }
-
-        @Override
-        public String toString()
-        {
-            return "CHUNKED_FIXED_SIZE";
-        }
-    };
+    NumberArrayFactory CHUNKED_FIXED_SIZE = new ChunkedNumberArrayFactory();
 
     /**
```

Comment:
I think we need to change a way how we keep exception in case if we fail to create array using all of the candidates, better keep each exception instead of overwriting it with the last one to see the actual reason instead of guessing it in case if all candidates will fail